tools/src/bin/fap-lld/main.rs

```diff
@@ -18,6 +18,18 @@ enum Error {
     ObjcopyFailed,
 }
 
+impl fmt::Display for Error {
```

Comment:
I suggest using `thiserror` here since it would correctly place `source`s and significantly reduce the amount of boilerplate.

Reply:
That's orthogonal to this change, which was to silence the clippy lint that the internal errors were not used. And in the tooling here, all that matters is the `Display` impl IIRC.

Reply:
Note that I'm not even implementing `std::error::Error`; sourcing etc is irrelevant in this case, as this code is not for a library (so the more appropriate dependency here if any would be `anyhow`).

Reply:
Good point